src/views/transactions/deposit/components/deposit-form/deposit-form.view.tsx

```diff
@@ -0,0 +1,123 @@
+import { AsyncTask, isAsyncTaskCompleted } from "src/utils/types";
+import { EstimatedL1Fee } from "src/domain";
+import SelectedAccount from "src/views/transactions/components/selected-account/selected-account.view";
+import DepositAlert from "src/views/transactions/deposit/components/deposit-alert/deposit-alert.view";
+import Fee from "src/views/transactions/components/fee/fee.view";
+import { AmountInputChangeEventData } from "src/views/shared/amount-input/amount-input.view";
+import TransactionAmountInput from "src/views/transactions/components/transaction-amount-input/transaction-amount-input.view";
+import PrimaryButton from "src/views/shared/primary-button/primary-button.view";
+
+export interface TxData {
+  amount: BigNumber;
+  from: EthereumAccount;
+}
+
+interface DepositFormStateProps {
+  account: EthereumAccount;
+  estimatedDepositFeeTask: AsyncTask<EstimatedL1Fee, Error>;
+  preferredCurrency: string;
+  fiatExchangeRatesTask: AsyncTask<FiatExchangeRates, string>;
+}
+
+interface DepositFormHandlerProps {
+  onGoToChooseAccountStep: () => void;
+  onSubmit: (txData: TxData) => void;
+}
+
+type DepositFormProps = DepositFormStateProps & DepositFormHandlerProps;
+
+const DepositForm: React.FC<DepositFormProps> = ({
+  account,
+  estimatedDepositFeeTask,
+  preferredCurrency,
+  fiatExchangeRatesTask,
+  onGoToChooseAccountStep,
+  onSubmit,
+}) => {
+  const classes = useDepositFormStyles();
+  const [amount, setAmount] = React.useState(BigNumber.from(0));
+  const [isAmountValid, setIsAmountValid] = React.useState<boolean | undefined>(undefined);
+  const [showInFiat, setShowInFiat] = React.useState(false);
+  const [areFundsExceededDueToFee, setAreFundsExceededDueToFee] = React.useState(false);
+
+  function isSubmitButtonDisabled() {
+    if (isAmountValid === false || areFundsExceededDueToFee) {
+      return true;
+    }
+
+    return false;
+  }
+
+  function handleAmountChange(data: AmountInputChangeEventData) {
+    setShowInFiat(data.showInFiat);
+    setAmount(data.amount.tokens);
+    setIsAmountValid(!data.isInvalid);
+    setAreFundsExceededDueToFee(data.areFundsExceededDueToFee);
+  }
+
+  function handleFormSubmission() {
+    onSubmit({
+      amount,
+      from: account,
+    });
+  }
+
+  return (
+    <FormContainer>
+      <SelectedAccount
+        txType={TxType.Deposit}
+        account={account}
+        preferredCurrency={preferredCurrency}
+        showInFiat={showInFiat}
+        fiatExchangeRatesTask={fiatExchangeRatesTask}
+        onClick={onGoToChooseAccountStep}
+      />
+      {areFundsExceededDueToFee && (
+        <DepositAlert
+          estimatedDepositFeeTask={estimatedDepositFeeTask}
+          preferredCurrency={preferredCurrency}
+          fiatExchangeRatesTask={fiatExchangeRatesTask}
+        />
+      )}
+      <form
+        className={classes.form}
+        onSubmit={(event) => {
+          event.preventDefault();
+          void handleFormSubmission();
```

Comment:
Why is this void required?

The same applies to `exit`, `force-exit`, and `transfer`

Reply:
It was used before as this functions were async. Removed in all the cases!